src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -0,0 +1,97 @@
+#include "RSUHealthMonitorPlugin.h"
+
+using namespace RSUHealthMonitor;
+using namespace tmx::utils;
+
+namespace RSUHealthMonitor
+{
+
+    RSUHealthMonitorPlugin::RSUHealthMonitorPlugin(const std::string &name) : PluginClient(name)
+    {
+        _rsuWorker = std::make_shared<RSUHealthMonitorWorker>();
+        _rsuStatusTimer = make_unique<ThreadTimer>();
+        UpdateConfigSettings();
+
+        // Send SNMP call to RSU periodically at configurable interval.
+        _timerThId = _rsuStatusTimer->AddPeriodicTick([this]()
+                                                      {
+            // Periodic SNMP call to get RSU status based on RSU MIB version 4.1
+            auto rsuStatusJson =  _rsuWorker->getRSUStatus(_rsuMibVersion, _rsuIp, _snmpPort, _securityUser, _authPassPhrase, _securityLevel, SEC_TO_MICRO);
```

Comment:
This is confusing to me. So you get the RSU status here by querying the SNMP server and you return a JSON. Then you wait until the broadcast method to check that the required information is present? You do not have to change this, it just seems weird to me. I would imagine that the method that gets the RSU status either directly turns it into the tmx message (including validation) or the next method turns it into a tmx message.

Reply:
If the JSON is empty due to some SNMP timeout or other issue, we stop any further logic like creating TMX message, broadcasting message.